File: app/src/main/java/com/romellfudi/callbacks/TemplateController.java
package com.romellfudi.callbacks;

/**
 * Created by romelldominguez on 9/30/16.
 */

public class TemplateController<G,H> {
    public void methodStatickCall(TemplateCallback<G,H> callback){
        callback.methodSuperReturnA(null);
        callback.methodPoorReturnY(null);
    }

    public static abstract class TemplateCallback<TC,UC>
            implements IntTemUnder<TC,UC>{

    }
}

File: app-kotlin/src/main/java/com/romellfudi/callbacks/TemplateController.kt
package com.romellfudi.callbacks

/**
 * Created by romelldominguez on 9/30/16.
 */
class TemplateController<G, H> {
    fun methodStatickCall(callback: TemplateCallback<G?, H?>) {
        callback.methodSuperReturnA(null)
        callback.methodPoorReturnY(null)
    }

    abstract class TemplateCallback<TC, UC> : IntTemUnder<TC, UC>
}